torch/_dynamo/eval_frame.py

```diff
@@ -462,7 +463,7 @@ def catch_errors(frame, cache_size, frame_state):
                     )
                     return hijacked_callback(frame, cache_size, hooks, frame_state)
 
-        with compile_lock:
+        with compile_lock, _disable_current_modes():
```

Comment:
For my learning why this change?

Reply:
Maybe @ezyang can explain this, but IIRC this helped in finding the right fake mode instance during tracing

Reply:
The basic concept is that when we are running the compiler stack, we don't actually want user modes to be turned on, they can interfere unpredictably with the stack. In this particular case, fake mode is causing the problem.

I am actually a bit surprised this passes all CI, because it is kind of risky, if we are depending on a different mode still being enabled internally for eg AOTAutograd...